### PR TITLE
fix frontend bugs

### PR DESCRIPTION
**Pull request by iamvigneshwars**

### CODE REVIEW

File: frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts [hunk 1]
```diff
@@ -61,27 +61,33 @@ export function generateNodesAndEdges(taskNodes: TaskNode[]): {
...
-        edges.push({
-          id: `e${parent}-${task.id}`,
-          source: parent,
-          target: task.id,
-          animated: true,
-        });
+        const edgeId = `e${parent}-${uniqueTaskId}`;
+        if (!edgeSet.has(edgeId)) {
+          edges.push({
+            id: edgeId,
```
Comment (garryod) [inline frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts:80]: Does this need to be a `string` or can it be a tuple?

File: frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts [hunk 2]
```diff
@@ -61,27 +61,33 @@ export function generateNodesAndEdges(taskNodes: TaskNode[]): {
...
+            source: parent,
+            target: uniqueTaskId,
```
Comment (garryod) [inline frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts:82]: Does this need to be a `string` or could it be a tuple?

File: frontend/workflows-lib/lib/components/workflow/TasksTable.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ const TasksTable: React.FC<TaskTableProps> = ({ tasks }) => {
   return (
     <Grid container spacing={1.5} padding={2} sx={{ overflow: "auto" }}>
       {tasks.map((task) => (
-        <Grid item key={task.name} xs="auto">
+        <Grid item key={task.id} xs="auto">
```
Comment (garryod) [inline frontend/workflows-lib/lib/components/workflow/TasksTable.tsx:15]: Do we want to use the unique id here too?

**Review by garryod — COMMENTED** [review]
Ah, those are a nice few gotchas

File: frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts (hunk 1 above)
Comment (iamvigneshwars) [reply frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts:80]: Needs to be string

File: frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts (hunk 2 above)
Comment (iamvigneshwars) [reply frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts:82]: Needs to be string

File: frontend/workflows-lib/lib/components/workflow/TasksTable.tsx (hunk 3 above)
Comment (iamvigneshwars) [reply frontend/workflows-lib/lib/components/workflow/TasksTable.tsx:15]: Yup, it complains!

File: frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts [hunk 4]
```diff
@@ -61,27 +61,33 @@ export function generateNodesAndEdges(taskNodes: TaskNode[]): {
 } {
   const nodes: Node[] = [];
   const edges: Edge[] = [];
+  const edgeSet = new Set<string>();
 
   const traverse = (tasks: TaskNode[], parents: string[] = []) => {
     tasks.forEach((task) => {
+      const uniqueTaskId = `${task.id}-${task.name}`;
```
Comment (garryod) [inline frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts:68]: If `task.id` is already unique this shouldn't be necessary

File: frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts [hunk 5]
```diff
@@ -61,27 +61,33 @@ export function generateNodesAndEdges(taskNodes: TaskNode[]): {
...
+        const edgeId = `e${parent}-${uniqueTaskId}`;
+        if (!edgeSet.has(edgeId)) {
+          edges.push({
+            id: edgeId,
+            source: parent,
+            target: uniqueTaskId,
+            animated: true,
+          });
+          edgeSet.add(edgeId);
+        }
```
Comment (garryod) [inline frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts:86]: Would it be nicer to construct a set of edges (i.e. `Set<Edge>`) then map it into an array (i.e. `Edge[]`)?

**Review by garryod — APPROVED** [review]
LGTM

**Pull request merged**